apis/services/userService.js

```diff
@@ -1,14 +1,32 @@
 const { userDao } = require('../models');
+const bcyrpt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const { validateUsername, validatePassword } = require('../utils/validators')
 
-// const signUp = async () => {
+const signIn = async (username, password) => {
 
-// }
+    validateUsername(username);
+    validatePassword(password);
 
-// const signIn = async () => {
+    const user = await userDao.getUserByUsername(username);
+    const passwordMatch = await bcyrpt.compare(password, user.password)
+    console.log(passwordMatch)
 
-// }
+    if(!passwordMatch) {
+        const error = new Error('WRONG_PASSWORD')
```

Comment:
로그인 에러처리 시에 ID가 틀렸는지, 비밀번호가 틀렸는지 모르게 하는 것이 좋습니다.
따라서 2개의 에러처리를 모두 같은 메시지로 통일하면 좋습니다.

```suggestion
    if(!passwordMatch) {
        const error = new Error('INVALID_USER')
```

Reply:
수정